src/network/server.cc

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2016 CodiLime
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "db/handle.h"
+#include "db/object.h"
+#include "dbif/types.h"
+#include "network/server.h"
+#include "util/settings/network.h"
+
+#include <QtNetwork/QTcpSocket>
+#include <QtEndian>
+#include <QDataStream>
+#include <QSettings>
+
+namespace veles {
+namespace db {
+
+NetworkServer::NetworkServer(PLocalObject root) :
+  root_(root), tcp_server_(new QTcpServer(this)) {
+  int32_t port = util::settings::network::port();
+  QHostAddress ip_addr(util::settings::network::ipAddress());
+  if (!tcp_server_->listen(ip_addr, port)) {
+    // TODO some error logging here
+    return;
+  }
+  connect(tcp_server_, &QTcpServer::newConnection, this, &NetworkServer::handle);
+}
+
+void NetworkServer::handle() {
+  QTcpSocket *client_connection = tcp_server_->nextPendingConnection();
+  connect(client_connection, &QAbstractSocket::disconnected,
+          client_connection, &QObject::deleteLater);
+  connect(client_connection, &QIODevice::readyRead, [this, client_connection] () {
+    readMessage(client_connection);
+  });
+}
+
+void NetworkServer::readMessage(QTcpSocket *client_connection) {
+  int32_t msg_len;
+  if (client_connection->bytesAvailable() < static_cast<int32_t>(sizeof(msg_len))) {
```

Comment:
Why is message length a signed integer? Someone can easily craft a packet with this being a negative number.